Fixes for 10.0

MyGamesDevelopmentAcc opened the pull request:
Wasn't making any updates to spells, just modified it so it works without errors. Did some testing, haven't noticied any errors.

Changes:
Changes required by new handling of cast bar. Kept back compatiblity with color setting as much as I could best to my knowledge

MyGamesDevelopmentAcc commented:
Hiding Blizzard frames does not yet work properly.

MyGamesDevelopmentAcc commented:
They seemto hide now, but I think there is still an issue with setting focus by mouse

MyGamesDevelopmentAcc commented:
The issue with focus and target is resolved.

jturel commented:
@MyGamesDevelopmentAcc let me know when you don't expect to push any more changes and I can test

MyGamesDevelopmentAcc commented:
I do not expect to push any more changes. To be honest I am not too fluent in githubish and didn't know initially that my 2nd commit through the github app will be attached automatically to this merge request, so I had to keep on rolling with it ;) Sorry about that.

jturel commented:
> I do not expect to push any more changes. To be honest I am not too fluent in githubish and didn't know initially that my 2nd commit through the github app will be attached automatically to this merge request, so I had to keep on rolling with it ;) Sorry about that.

no worries. pushing to the branch that you created this PR from also updates the PR!

jturel commented on sArena.toc:
```diff
@@ -1,8 +1,8 @@
 ## Notes: Enhanced arena frames
 ## Author: Stako
-## Version: 3.1.8
+## Version: 3.1.9
```
Make this 3.2 to signify a more significant change (backward incompatible with old version)

jturel commented on sArena.xml:
```diff
@@ -106,14 +107,16 @@
-        </Scripts>
         <BarTexture file="Interface\TargetingFrame\UI-StatusBar"/>
         <BarColor r="1.0" g="0.7" b="0.0"/>
+		<Scripts>
```
Tab indentation

jturel commented on sArena.lua:
```diff
@@ -578,7 +580,7 @@ function sArenaFrameMixin:ResetTrinket()
 
 local function ResetStatusBar(f)
-    f:SetStatusBarTexture(nil)
+    --f:SetStatusBarTexture(nil)
```
What do you think about just deleting this line?

jturel commented on sArena.lua:
```diff
@@ -294,10 +294,12 @@ function sArenaFrameMixin:OnLoad()
     self.unit = unit
 
-    CastingBarFrame_SetUnit(self.CastBar, unit, false, true)
+	self.CastBar:SetUnit(unit, false, true)
```
Tab

jturel commented on sArena.lua:
```diff
@@ -63,7 +63,7 @@ local UnitFrameHealPredictionBars_Update = UnitFrameHealPredictionBars_Update
     -- hide blizz arena frames while in arena
     if ( InCombatLockdown() ) then return end
-    if ( not IsAddOnLoaded("Blizzard_ArenaUI") ) then return end
+    --if ( not IsAddOnLoaded("Blizzard_ArenaUI") ) then return end
```
Delete this, or keep?

jturel commented on Layouts/Xaryu.lua:
```diff
@@ -229,6 +229,11 @@ end
 function layout:UpdateTextures(frame)
     local texture = self.db.classicBars and "Interface\\TargetingFrame\\UI-StatusBar" or "Interface\\RaidFrame\\Raid-Bar-Hp-Fill"
 
+	frame.CastBar.typeInfo = {
```
Tab indentation

jturel commented on Layouts/BlizzTourney.lua:
```diff
@@ -123,7 +123,13 @@ function layout:Initialize(frame)
-    f:SetStatusBarTexture("Interface\\RaidFrame\\Raid-Bar-Hp-Fill")
+	local typeInfoTexture = "Interface\\RaidFrame\\Raid-Bar-Hp-Fill";
+    f:SetStatusBarTexture(typeInfoTexture)
+	f.typeInfo = {
```
Tab indentation

jturel commented on Layouts/BlizzTarget.lua:
```diff
@@ -128,7 +128,13 @@ function layout:Initialize(frame)
-    f:SetStatusBarTexture("Interface\\TargetingFrame\\UI-StatusBar")
+	local typeInfoTexture = "Interface\\TargetingFrame\\UI-StatusBar";
+    f:SetStatusBarTexture(typeInfoTexture)
+	f.typeInfo = {
```
Tab indentation

jturel commented on Layouts/BlizzArena.lua:
```diff
@@ -127,7 +127,13 @@ function layout:Initialize(frame)
 
     f = frame.CastBar
-    f:SetStatusBarTexture("Interface\\TargetingFrame\\UI-StatusBar")
+	local typeInfoTexture = "Interface\\TargetingFrame\\UI-StatusBar";
```
Tab indentation

jturel reviewed (changes requested):
Thank you @MyGamesDevelopmentAcc - this is a big improvement!

I left a few comments - mostly around using tab indentation vs space indentation. Better to be consistent with what's already there :)

MyGamesDevelopmentAcc commented:
@jturel I have made the requested fixes, added some further formatting at the end, fixed further problems found with WoW10 handling of SecureActionButtons in certain setups.

jturel commented:
> I am also thinking that this will most likely break, if there was, the compatiblity with older Wow versions as I have not added any special handling to distinguish whether this is wow10 or not.

As far as I'm aware the "retail" version of sArena has supported only the latest wow version!

jturel commented:
This got a lot harder to review because of all the extra formatting changes that got included. I'll need some time :)

MyGamesDevelopmentAcc commented:
For that reason I have done it in separate commits, hoping it will be easier to see which was changed/modified in each.

jturel commented:
@MyGamesDevelopmentAcc i fixed an automatic action that should run in this repo to validate the lua syntax automatically. Can you do me a favor and rebase your branch against the master branch? After you push, you'll see the job run here and tell you whether it passes or fails.

Your steps would look something like the below, given a remote called 'origin' corresponding to this repo (not your fork):

```
git fetch origin
git rebase origin/master
git push $YOUR_FORK_REMOTE $YOUR_BRANCH_NAME --force
```

MyGamesDevelopmentAcc commented:
Added two new globals and rebased. Please let me know if this is what you asked for.

jturel commented on .luacheckrc:
```diff
@@ -49,5 +49,7 @@ read_globals = {
     'UnitPowerMax',
     'UnitPowerType',
     'UnitRace',
+    'ArenaEnemyFramesContainer',
```
This looks good. Can you put these at the top of the file? They seem to be in alphabetical order :)

jturel commented:
Thank you @MyGamesDevelopmentAcc ! Unfortunately this is as far as I can help you as I don't have the ability to create a new release on Curse. Merging this.

MyGamesDevelopmentAcc commented:
@stako any chances for you to release? @jturel maybe we can add release format which could be picked by wowup?

jturel commented:
> @stako any chances for you to release this please? @jturel maybe we can add release format which could be picked by wowup?

hey there @MyGamesDevelopmentAcc - missed this message. I'm happy to review and test any changes to get the addon picked up by wowup :)
